src/HasBinaryUuid.php

```diff
@@ -73,7 +73,54 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        return array_merge(parent::toArray(), [$this->getKeyName() => $this->uuid_text]);
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+        foreach ($uuidAttributes as $attributeKey) {
+            if (array_key_exists($attributeKey, $array)) {
+                $array[$attributeKey] = $this->{"{$attributeKey}_text"};
+            }
+        }
+
+        return $array;
+    }
+
+    public function getUuidAttributes()
+    {
+        $uuidAttributes = [$this->getKeyName()];
+        if (property_exists($this, 'uuidAttributes')) {
+            $uuidAttributes = $this->uuidAttributes === null ? [] : $this->uuidAttributes;
+        }
+
+        return $uuidAttributes;
+    }
+
+    public function getAttribute($key)
+    {
+        if (($uuidKey = $this->uuidTextAttribute($key)) && $this->{$uuidKey} !== null) {
```

Comment:
While these shorthand assignments are useful for easy writing, it makes it a lot more difficult to read the code. I'd prefer an extra assignment above this `if` statement for clarity.